### PR TITLE
Don't update Java minimum memory based on maximum memory allocation

### LINKED ISSUE #427: Quick Setup: Do not change ‘Minimum memory’ based on ‘Maximum memory’ field
### Operating System

Windows, macOS, Linux, Other

### Version of Prism Launcher

Prism Launcher 5.1

### Version of Qt

Qt 6.4.0

### Description of bug

When entering into the field ‘Maximum memory allocation’ a number less than the number in ‘Minimum memory allocation’, the number in ‘Minimum memory allocation’ automatically updates to the number in ‘Maximum memory allocation’, creating frustrating experience where you have to go back and change value in ‘Minimum memory allocation’ again.

Would be better if this updating behaviour would only happen when ‘Maximum memory allocation’ field lost focus. Ideally, however, it should not happen *at all*, instead showing an error when user clicks the ‘Finish’ button that their ‘Maximum memory allocation’ is less than value in ‘Minimum memory allocation’, forcing them to notice and manually fix the mistake. The reason for the error rather than guess work is because this may slip through unnoticed, especially if the user relies on assistive technologies like screen reader that won't announce that ‘Minimum memory allocation’ value has been automatically updated.

### Steps to reproduce

1. Remove launcher configuration / start fresh installation to force quick setup to show.
2. Select your language to proceed to problematic step.
3. On Java step enter ‘256 (MiB)’ in ‘Minimum memory allocation’.
4. Select everything in ‘Maximum memory allocation’ field and start entering ‘2048 (MiB)’ one digit at a time.

Observe as you enter the digit 4, that the value in field ‘Minimum memory allocation’ changes to match what's currently entered in ‘Maximum memory allocation’ field, thus to ‘204 (MiB)’.

GIF showing the author performing the aforementioned reproduce steps

### Suspected cause

Poor design decisions.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #427
Removed the code that modifies the user values.

### CODE REVIEW

**Review by TayouVR — APPROVED** [review]
it does fix the issue. 
However.. when the value is invalid the icon on the java check button changes. this already was the case before, it would just immediately change back. 
I don't think that should be though. Instead the little icon next to the input field should change maybe; not the java check button icon

**Comment by Trial97** [comment]
The Java check will still flicker as each time we update the memory we check if Java is valid, with the new memory.
Now also added a warning for max memory if the min value is greater.

**Review by TayouVR — APPROVED** [review]

**Comment by Scrumplex** [comment]
> Now also added a warning for max memory if the min value is greater.

Could you also do the same for the global settings?

**Comment by Trial97** [comment]
> > Now also added a warning for max memory if the min value is greater.
> 
> Could you also do the same for the global settings?

Yes

File: launcher/ui/pages/global/JavaPage.cpp [hunk 1]
```diff
@@ -194,6 +195,9 @@ void JavaPage::updateThresholds()
     } else if (maxMem > (sysMiB * 0.9)) {
         iconName = "status-yellow";
         ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation approaches your system memory capacity."));
+    } else if (maxMem < minMem) {
+        iconName = "status-yellow";
+        ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller that the minimum value"));
```
Comment (TayouVR) [inline launcher/ui/pages/global/JavaPage.cpp]:
```suggestion
        ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller than the minimum value"));
```

File: launcher/ui/pages/instance/InstanceSettingsPage.cpp [hunk 2]
```diff
@@ -487,6 +488,9 @@ void InstanceSettingsPage::updateThresholds()
     } else if (maxMem > (sysMiB * 0.9)) {
         iconName = "status-yellow";
         ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation approaches your system memory capacity."));
+    } else if (maxMem < minMem) {
+        iconName = "status-yellow";
+        ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller that the minimum value"));
```
Comment (TayouVR) [inline launcher/ui/pages/instance/InstanceSettingsPage.cpp]:
```suggestion
        ui->labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller than the minimum value"));
```

File: launcher/ui/widgets/JavaSettingsWidget.cpp [hunk 3]
```diff
@@ -415,6 +415,9 @@ void JavaSettingsWidget::updateThresholds()
     } else if (observedMaxMemory > (m_availableMemory * 0.9)) {
         iconName = "status-yellow";
         m_labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation approaches your system memory capacity."));
+    } else if (observedMaxMemory < observedMinMemory) {
+        iconName = "status-yellow";
+        m_labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller that the minimum value"));
```
Comment (TayouVR) [inline launcher/ui/widgets/JavaSettingsWidget.cpp]:
```suggestion
        m_labelMaxMemIcon->setToolTip(tr("Your maximum memory allocation is smaller than the minimum value"));
```

**Review by TayouVR — COMMENTED** [review]
overlooked these typos, oops

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**